### PR TITLE
Add more aliasing support

### PR DESCRIPTION
**Pull request by terrycain**
An MR to fix a few things, will tag people to check over some stuff in a bit.

### CODE REVIEW

**Comment by terrycain** [comment]
If people are able to test this without me having to do a release, I'd appreciate it. Am not finished but this branch currently should be in a working state.

@snigel - This should fix your issues on #30. I reckon it can still be improved more, but it should do until then.
@mdhowle - Have overhauled the DNS delegation docs section and added some more features, you mind having a read over it to see if it makes sense?
@theodiem - You mentioned in #20 about giving certbot only permissions to specific records and not deleting them during cleanup, this allows this somewhat. Currently you'll need to add an entry for each record you wish to not delete. If I can summon the will to redo how the config works, I might look into making this much nicer.

**Review by mdhowle — COMMENTED** [review]
I haven't had time to look at the code portion, but the documentation looks okay to me.

If you want to give some external resources, these links could be helpful:

- https://letsencrypt.org/docs/challenge-types/#dns-01-challenge  (fourth paragraph)
- https://www.eff.org/deeplinks/2018/02/technical-deep-dive-securing-automation-acme-dns-challenge-validation

File: certbot_dns_azure/__init__.py [hunk 1]
```diff
@@ -120,30 +120,84 @@
...
 DNS delegation
 --------------
+
 DNS delegation, also known as DNS aliasing, is a process of allowing a secondary DNS zone to handle validation in place
 of the primary zone. For example, you would like to acquire a certificate for ``example.com`` but have the validation
 performed on a secondary domain ``example.org``. You would create a ``_acme-challenge.example.com`` CNAME on the
-``example.com`` nameserver with the value of ``_acme-challenge.example.org``. Certbot will resolve the CNAME and
-validate the ``example.com`` domain.
+``example.com`` nameserver with the value of ``someotherrecord.example.org``. The ACME server will resolve the CNAME and
+validate the TXT record ``someotherrecord.example.org`` instead. Certbot itself does not support CNAME aliasing,
```
[the remark this answers is not in the document]
Comment (terrycain) [reply certbot_dns_azure/__init__.py]: Thanks :smile:

File: certbot_dns_azure/__init__.py [hunk 2]
```diff
@@ -120,30 +120,84 @@
...
+therefore this plugin does what it can to support it.
 
 The common reasons for DNS delegation are:
- * The primary DNS zone is hosted on a nameserver with no API access
- * Security concerns regarding access to the primary DNS zone
+ #. The primary DNS zone is hosted on a nameserver with no API access
+ #. Security concerns regarding access to the primary DNS zone
```
[the remark this answers is not in the document]
Comment (terrycain) [reply certbot_dns_azure/__init__.py]: Thanks :smile:

File: certbot_dns_azure/__init__.py [hunk 3]
```diff
@@ -120,30 +120,84 @@
...
+
+Example: Primary Zone, no API Access
+++++++++++++++++++++++++++++++++++++
 
-To use DNS delegation:
- #. Manually create the ``_acme-challenge.<primary domain>`` CNAME with the value ``_acme-challenge.<secondary domain>`` on the ``example.com`` nameserver.
- #. In the certbot azure configuration file, specify the primary domain and the entire secondary DNS zone's resource ID in ``dns_azure_zoneX``.
- #. Request the certificate.
+Let's assume you wish to get a certificate for ``test.foo.com``, this will result in a validation record for
+``_acme-challenge.test.foo.com``. Assuming you don't have access to the ``foo.com`` zone, you can CNAME said
```
[the remark this answers is not in the document]
Comment (terrycain) [reply certbot_dns_azure/__init__.py]: Thanks :smile:

**Comment by terrycain** [comment]
@snigel @mdhowle @theodiem These changes are now out on 2.2.0b0 on PyPI and on the snap edge channel. Assuming I get no complaints, I'll release 2.2.0 officially in a week or so.

**Pull request merged**